### PR TITLE
Add dependencies command

### PR DESCRIPTION
**Pull request by maurelian**

### CODE REVIEW

File: src/dependencies.js [hunk 1]
```diff
@@ -44,11 +49,28 @@ export function dependencies(files) {
...
   }
 
-  dependencies = linearize(dependencies, {reverse: true})
+  if (!dependencies[childContract]) {
+    console.log('Specified child contract not found. Bailing.. ')
+  }
 
-  for (let contractDependencies of dependencies) {
-    console.log(contractDependencies[0].orange)
+  // the c3-linearize package's reverse feature doesn't seem to work for deeper levels of inheritance
```
[the remark this answers is not in the document]
Comment (maurelian) [reply src/dependencies.js]: So, I looked closer, and I think the issue is that linearization isn't deterministic (afaict). ie. you can have various orderings, as long as nothing is listed before one of its own ancestors.

But I found the ordering looked quite odd when I reversed it using c3-linearization, as opposed to reversing the input before hand. 

For example, I ran it on the [0x Exchange.sol contract](https://github.com/0xProject/0x-monorepo/blob/development/packages/contracts/src/2.0.0/protocol/Exchange/Exchange.sol):

```solidity
contract Exchange is
    MixinExchangeCore,
    MixinMatchOrders,
    MixinSignatureValidator,
    MixinTransactions,
    MixinAssetProxyDispatcher,
    MixinWrapperFunctions
```

**Reversed before linearizing:**

`Exchange
└─ MixinWrapperFunctions <- MixinAssetProxyDispatcher <- MixinTransactions <- MixinSignatureValidator <- MixinMatchOrders <- MixinExchangeCore <- MTransactions <- ITransactions <- MSignatureValidator <- ISignatureValidator <- MMatchOrders <- IMatchOrders <- MExchangeCore <- IExchangeCore <- LibAbiEncoder <- MAssetProxyDispatcher <- IAssetProxyDispatcher <- LibFillResults <- Ownable <- IOwnable <- LibOrder <- LibEIP712 <- LibMath <- SafeMath <- LibConstants`

**Reversed by the c3-linearization package:**

`
Exchange
└─ MixinExchangeCore <- MixinMatchOrders <- LibConstants <- MixinSignatureValidator <- MixinTransactions <- MixinAssetProxyDispatcher <- Ownable <- IOwnable <- MixinWrapperFunctions <- LibMath <- LibOrder <- LibEIP712 <- LibFillResults <- SafeMath <- MAssetProxyDispatcher <- IAssetProxyDispatcher <- LibAbiEncoder <- MExchangeCore <- IExchangeCore <- MSignatureValidator <- ISignatureValidator <- MMatchOrders <- IMatchOrders <- MTransactions <- ITransactions
`

[the remark this answers is not in the document]
Comment (GNSPS) [reply src/dependencies.js]: Just checked it and I get an ordering that is not agreeing with Solidity's example of inheritance.

I ran the inheritance contract example from Solidity's docs and I get the same result for the example contract's inheritance that they reference there:
```
Final
└─ Base2 <- Base1 <- mortal <- owned
```

**Comment by GNSPS** [comment]
Would you be able to create a README entry for the command, as well, @maurelian? 😄

File: src/dependencies.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (GNSPS) [reply src/dependencies.js]: Can you check your package-lock.json for your current version of the c3-linearization module? Is it 0.3.0?
Maybe that's it! 😅

File: bin/surya [hunk 2]
```diff
@@ -31,14 +31,18 @@ require('yargs') // eslint-disable-line
   }, (argv) => {
     inheritance(argv.files)
   })
-  .command('dependencies <files..>', 'output a linearized list of smart contract dependencies (linerized inherited parents).', (yargs) => {
+  .command('dependencies <child_contract> <files..>', 'output a linearized list of smart contract dependencies (linerized inherited parents).', (yargs) => {
```
[the remark this answers is not in the document]
Comment (GNSPS) [reply bin/surya]: Also, could you match this new var naming inside `dependencies.js`?

**Comment by maurelian** [comment]
image

^ just for the hosting

**Comment by GNSPS** [comment]
LGTM! Merging...

**Pull request merged**